### PR TITLE
Feature/4840 fe validation

### PR DESCRIPTION
**Pull request by alibombaye**
Story [AB#4840](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4840)
Task [AB#7100](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7100)

Screenshot 2020-05-20 at 17 32 49

### CODE REVIEW

File: app/pages/sections/supplier/search/template.test.js [hunk 1]
```diff
@@ -23,6 +23,31 @@ describe('supplier search page', () => {
...
+      const errorArray = $('[data-test-id="error-summary"] li a');
+      expect(errorSummary.length).toEqual(1);
+      expect(errorArray.length).toEqual(context.errors.length);
+      context.errors.forEach((error, i) => {
+        expect(errorArray[i].attribs.href).toEqual(error.href);
+        expect(errorArray[i].children[0].data.trim()).toEqual(error.text);
+      });
+    });
+  }));
+
```
Comment (kat1906) [inline app/pages/sections/supplier/search/template.test.js:50]: do we need tests for error messages around the text input field?

File: app/routes.js [hunk 2]
```diff
@@ -92,6 +92,18 @@ export const routes = (authProvider) => {
...
+  router.post('/organisation/:orderId/supplier/search', authProvider.authorise({ claim: 'ordering' }), withCatch(authProvider, async (req, res) => {
+    const { orderId } = req.params;
+
+    const response = validateSupplierSearchForm({ data: req.body });
+
+    const context = await getSupplierSearchPageErrorContext({
+      orderId,
+      validationErrors: response.errors,
+    });
+    return res.render('pages/sections/supplier/search/template.njk', addContext({ context, user: req.user, csrfToken: req.csrfToken() }));
```
Comment (kat1906) [inline app/routes.js:104]: are you doing the happy path in a separate ticket? ie `success: true`

Comment (alibombaye) [reply app/routes.js:104]: 👍 yes

File: app/pages/sections/supplier/search/template.test.js (hunk 1 above)
Comment (alibombaye) [reply app/pages/sections/supplier/search/template.test.js:50]: Cool I'll add something around this 👍

**Pull request merged**